[fix] Non interactive modes ( [Bug]: sanitize_input #324 )

Tearran opened the pull request:
Issues 
- #324 
- #322 
- #325 

Staging for refining `Non interactive` modes. 
added option "--noint" 
stage "--api" for json I/O (currently only out). 
Remove calls for `sanitize_input` in main binary options,

dimitry-ishenko commented:
@Tearran I will take a closer look when I get a chance, but on the first peek I was thinking maybe we can add `have_stdin` and `have_stdout` helpers or something similar to replace all the `[[ -t 0 ]]` and `[[ -t 1 ]]`. I've had a couple of them in my PRs and see a bunch more in yours.

Tearran commented:
Thanks. I support the use of `have_stdin and have_stdout`. but consider `set_stdin set_stdout` or  get_  :)

Tearran commented:
I reverted armbian-config and moved the changes to armbian-config-dev. 

This setup could allow us to stage and refine these changes. 

Staging changes may benefit the project by letting us iteratively improve features, naming conventions, and json keys and functionalities. 

This approach should help ensure that aspects which might be incomplete or not fully optimized can be refined in stages.

Tearran commented:
added timer function  to help identify sections that are slow and may need optimization or separation to improve performance. may help pinpoint where the script might be experiencing delays and take appropriate actions to optimize those areas

dimitry-ishenko reviewed (approved):
@Tearran this looks cool. Are you planning to merge this back into `armbian-config` or keep it as a separate tool?

Tearran commented:
>Are you planning to merge this back into armbian-config or keep it as a separate tool?

Conceptually to end up as armbian-config while setting precedence for a development binary, in other words either way and/or both will work for me :) 

I believe I already have the module worked out for this `https://github.com/Tearran/configng/blob/Tearran-patch-5/tools/dev/message_checkpoint.sh`  

lets see if I can clean up these conflicts.

Tearran commented:
@dimitry-ishenko refined it here https://github.com/armbian/configng/pull/405
